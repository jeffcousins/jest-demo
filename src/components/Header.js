import React, { Component } from 'react';

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: 'OMG I FCKING LOVE JEST'
    };
  }

  render() {
    return (
      <h2>{this.state.text}</h2>
    );
  }
}

export default Header;
