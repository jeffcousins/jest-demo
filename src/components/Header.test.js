import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

it('should render the header', () => {
  const component = renderer
    .create(<Header />);

  const domTree = component.toJSON();

  expect(domTree).toMatchSnapshot();
})
