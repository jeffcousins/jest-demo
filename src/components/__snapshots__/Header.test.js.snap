exports[`test should render the header 1`] = `
<h2>
  OMG I FCKING LOVE JEST
</h2>
`;
